src/util/AbstractHelper.sol

```diff
@@ -0,0 +1,337 @@
+
+    /**
+    @notice Borrows on behalf of the caller, buying the necessary DBR to pay for the loan over the period, by borrowing aditional funds to pay for the necessary DBR
+    @dev Has to borrow the maxDebt amount due to how the market's borrowOnBehalf functions, and repay the excess at the end of the call resulting in a weird repay event
+    @param market Market the caller wishes to borrow from
+    @param dolaAmount Amount the caller wants to end up with at their disposal
+    @param maxDebt The max amount of debt the caller is willing to end up with
+    @param duration The duration the caller wish to borrow for
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function buyDbrAndBorrowOnBehalf(
+        IMarket market, 
+        uint dolaAmount,
+        uint maxDebt,
+        uint duration,
+        uint deadline, 
+        uint8 v, 
+        bytes32 r, 
+        bytes32 s) 
+        public 
+    {
+        //Calculate DOLA needed to pay out dolaAmount + buying enough DBR to approximately sustain loan for the duration
+        (uint dolaToBorrow, uint dbrNeeded) = approximateDolaAndDbrNeeded(dolaAmount, duration, 8);
+        require(maxDebt >= dolaToBorrow, "Cost of borrow exceeds max borrow");
+
+        //Borrow Dola
+        market.borrowOnBehalf(msg.sender, maxDebt, deadline, v, r, s);
+        
+        //Buy DBR
+        _buyExactDbr(dbrNeeded, maxDebt - dolaAmount);
+
+        //Transfer remaining DBR and DOLA amount to user
+        DOLA.transfer(msg.sender, dolaAmount);
+        DBR.transfer(msg.sender, DBR.balanceOf(address(this)));
+
+        //Repay what remains of max borrow
+        uint dolaBalance = DOLA.balanceOf(address(this));
+        DOLA.approve(address(market), dolaBalance);
+        market.repay(msg.sender, dolaBalance);
+    }
+
+    /**
+    @notice Deposits collateral and borrows on behalf of the caller, buying the necessary DBR to pay for the loan over the period, by borrowing aditional funds to pay for the necessary DBR
+    @dev Has to borrow the maxDebt amount due to how the market's borrowOnBehalf functions, and repay the excess at the end of the call resulting in a weird repay event
+    @param market Market the caller wish to deposit to and borrow from
+    @param dolaAmount Amount the caller wants to end up with at their disposal
+    @param maxDebt The max amount of debt the caller is willing to end up with
+    @param duration The duration the caller wish to borrow for
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function depositBuyDbrAndBorrowOnBehalf(
+        IMarket market, 
+        uint collateralAmount, 
+        uint dolaAmount,
+        uint maxDebt,
```

Comment:
`maxDebt` is sensitive. UI should provide a reasonable value to prevent sandwich attacks. Comments should make note of it for users who interact directly with the contracts.

Reply:
Fixed in [74e7e73](https://github.com/InverseFinance/FiRM/pull/33/commits/74e7e733169f454ecae4b9c771f033b0cdad8d84)